core/lib/zksync_core/src/eth_sender/eth_tx_aggregator.rs

```diff
@@ -436,87 +436,31 @@ impl EthTxAggregator {
 
         let (calldata, sidecar) = match op.clone() {
             AggregatedOperation::Commit(last_committed_l1_batch, l1_batches, pubdata_da) => {
-                let commit_data = self.l1_commit_data_generator.l1_commit_batches(
+                let commit_data_base = self.l1_commit_data_generator.l1_commit_batches(
                     &last_committed_l1_batch,
                     &l1_batches,
                     &pubdata_da,
                 );
-                if contracts_are_pre_shared_bridge {
-                    if let PubdataDA::Blobs = self.aggregator.pubdata_da() {
-                        let calldata = self
-                            .functions
-                            .pre_shared_bridge_commit
-                            .encode_input(&commit_data)
-                            .expect("Failed to encode commit transaction data");
-
-                        let side_car = l1_batches[0]
-                            .header
-                            .pubdata_input
-                            .clone()
-                            .unwrap()
-                            .chunks(ZK_SYNC_BYTES_PER_BLOB)
-                            .map(|blob| {
-                                let kzg_info = KzgInfo::new(blob);
-                                SidecarBlobV1 {
-                                    blob: kzg_info.blob.to_vec(),
-                                    commitment: kzg_info.kzg_commitment.to_vec(),
-                                    proof: kzg_info.blob_proof.to_vec(),
-                                    versioned_hash: kzg_info.versioned_hash.to_vec(),
-                                }
-                            })
-                            .collect::<Vec<SidecarBlobV1>>();
-
-                        let eth_tx_sidecar = EthTxBlobSidecarV1 { blobs: side_car };
-                        (calldata, Some(eth_tx_sidecar.into()))
-                    } else {
-                        let calldata = self
-                            .functions
-                            .pre_shared_bridge_commit
-                            .encode_input(&commit_data)
-                            .expect("Failed to encode commit transaction data");
-                        (calldata, None)
-                    }
+                let (encoding_fn, commit_data) = if contracts_are_pre_shared_bridge {
+                    (&self.functions.pre_shared_bridge_commit, commit_data_base)
                 } else {
-                    args.extend(commit_data);
-                    if let PubdataDA::Blobs = self.aggregator.pubdata_da() {
-                        let calldata = self
-                            .functions
+                    args.extend(commit_data_base);
```

Comment:
nit: I would also put `let mut args = vec![Token::Uint(self.rollup_chain_id.as_u64().into())];` inside `else` branch

Reply:
we need args in the other aggregated op branches as well (unless we want to duplicate code there)